gordon/record_checker.py

```diff
@@ -0,0 +1,112 @@
+import asyncio
+import logging
+import time
+
+from async_dns import types
+from async_dns.resolver import ProxyResolver
+
+
+class RecordChecker(object):
+    QUERYING_THRESHOLD = 60
+
+    def __init__(self, dns_ip):
+        """Setup RecordChecker object
+
+        Args:
+            dns_ip: DNS ip server to query
+        """
+        self._dns_ip = dns_ip
```

Comment:
What about the possibility to query multiple upstream resolvers? I could see the case where we want to know how long a record takes to resolve on a set of resolvers, not just one. We do this currently internally with dns-metrics.

Reply:
Also, should we check if the IP is valid before attempting to query it?

Reply:
so following our conversion, we will leave it to query just one DNS server for now..